File: Difficulty: Medium/k largest elements/k-largest-elements.py
import heapq

class Solution:
	def kLargest(self, arr, k):
	    max_heap = []
		for x in arr:
		    heapq.heappush(max_heap, -x)
		
		max_arr = []
		for i in range(0, k):
		    x = -heapq.heappop(max_heap)
		    max_arr.append(x)
		
		return max_arr
		    

File: Difficulty: Medium/k largest elements/k-largest-elements.java
class Solution {
    public ArrayList<Integer> kLargest(int[] arr, int k) {
        //Arrays.sort(arr);
        
        int n = arr.length;
        PriorityQueue<Integer> pq = new PriorityQueue<>(n);
        
        for (int x : arr) {
            pq.add(-x);
        }

        ArrayList<Integer> res = new ArrayList<>(k);
        for (int i = 0; i < k; i++) {
            if (!pq.isEmpty()) {
                int x = pq.poll();
                res.add(-x);
            }
        }
        
        return res;
    }
}
